File: ios/CoChat/Repositories/AuthRepository.swift
import Foundation

final class AuthRepository {
    private let api: APIClient
    private let tokenStore: TokenStore

    init(api: APIClient, tokenStore: TokenStore) {
        self.api = api
        self.tokenStore = tokenStore
    }

    var isLoggedIn: Bool { tokenStore.isLoggedIn }

    func register(fullName: String, email: String?, mobile: String?, password: String, designation: String?) async throws {
        let body = RegisterRequest(fullName: fullName, email: email, mobile: mobile, password: password, designation: designation)
        let res: AuthResponse = try await api.post("auth/register", body: body)
        tokenStore.save(accessToken: res.accessToken, refreshToken: res.refreshToken, user: res.user)
    }

    func login(identifier: String, password: String) async throws {
        let res: AuthResponse = try await api.post("auth/login", body: LoginRequest(identifier: identifier, password: password))
        tokenStore.save(accessToken: res.accessToken, refreshToken: res.refreshToken, user: res.user)
    }

    func logout() async {
        if let refreshToken = tokenStore.refreshToken {
            try? await api.postVoid("auth/logout", body: RefreshRequest(refreshToken: refreshToken))
        }
        tokenStore.clear()
    }
}

File: android/app/src/main/java/com/cochat/app/data/repository/AuthRepository.kt
package com.cochat.app.data.repository

import com.cochat.app.data.model.User
import com.cochat.app.data.remote.ApiService
import com.cochat.app.data.remote.LoginRequest
import com.cochat.app.data.remote.RefreshRequest
import com.cochat.app.data.remote.RegisterRequest
import com.cochat.app.data.remote.TokenStore
import kotlinx.coroutines.flow.Flow

class AuthRepository(private val api: ApiService, private val tokenStore: TokenStore) {

    val currentUser: Flow<User?> = tokenStore.userFlow
    val isLoggedIn: Boolean get() = tokenStore.isLoggedInBlocking()

    suspend fun register(fullName: String, email: String?, mobile: String?, password: String, designation: String?) {
        val res = api.register(RegisterRequest(fullName, email, mobile, password, designation))
        tokenStore.save(res.accessToken, res.refreshToken, res.user)
    }

    suspend fun login(identifier: String, password: String) {
        val res = api.login(LoginRequest(identifier, password))
        tokenStore.save(res.accessToken, res.refreshToken, res.user)
    }

    suspend fun logout() {
        val refreshToken = tokenStore.refreshTokenBlocking()
        if (refreshToken != null) {
            runCatching { api.logout(RefreshRequest(refreshToken)) }
        }
        tokenStore.clear()
    }
}
